pkg/sfu/redreceiver.go

```diff
@@ -131,6 +131,16 @@ func (r *RedReceiver) encodeRedForPrimary(pkt *rtp.Packet, redPayload []byte) (i
 }
 
 func encodeRedForPrimary(redPkts []*rtp.Packet, primary *rtp.Packet, redPayload []byte) (int, error) {
+	payloadSize := len(primary.Payload) + 1
+	for _, p := range redPkts {
+		payloadSize += len(p.Payload) + 4
+	}
+
+	// if required payload size is larger than the redPayload buffer, encode the primary packet only
+	if payloadSize > len(redPayload) {
+		redPkts = redPkts[:0]
```

Comment:
Another factor to consider is also bit rate. If the audio is already going at high bit rate (say 256 kbps or higher), using RED could bring it up to video rates. Something to think about there as well.

Reply:
we will close red if stereo is enabled, user can enable red encoding when publishing the track if he knows what he is doing and need.

Reply:
> For later. I think we are doing two packets redundancy now. Maybe, we can check how many redundant packets will fit and do that many (up to a maximum of 2) instead of just encoding the primary packet?

That makes sense, it can be a later optimize